cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java

```diff
@@ -273,7 +273,14 @@ public String toString() {
         RestartContext ctx = podToContext.computeIfAbsent(podId,
             k -> new RestartContext(backoffSupplier));
         singleExecutor.schedule(() -> {
+            if (reconciliation.isCancelled()) {
```

Comment:
Using the `Reconciliation` for checking for cancellation seems reasonable. But I think we should encapsulate the checking within `Reconcilation` with a `checkForCancellation` method, which throws a `ReconciliationCancelledException`. Within `KafkaRoller` we should be able to make that exit the rolling immediately, most likely propagating the/a `ReconciliationCancelledException` to the `KafkaAssemblyOperator` itself.

Reply:
Just to clarify - `checkForCancellation` would be a drop-in replacement for the current `isCancelled` calls, correct? Of course, with the logging/promise logic removed. I'm also assuming that `ReconciliationCancelledException` would be best as a `RuntimeException`.